### ISSUE TITLE
add support for printing maps in templates

### ISSUE THREAD

**Issue by brigadier** [erlydtl/erlydtl#228]
At the moment {{ map.key }} works, but {{ map }} doesn't, though it would be useful for debuging purposes

**Comment by seriyps** [comment]
So, what you expect as output and what you get now?

**Comment by brigadier** [comment]
I expect io_lib:format("~p", [Map]).
Now i get an exception.

**Comment by kaos** [comment]
Try `{{ map | pprint }}`
https://github.com/erlydtl/erlydtl/blob/master/src/erlydtl_filters.erl#L532

**Comment by brigadier** [comment]
Well, pprint is ok, but it possible to show both lists and tuples, so why not maps?

I've looked on the `stringify_final` function in the `erlydtl_runtime.erl` and it skips type conversion altogether for everything but lists, atoms and tuples. I think funs, floats and ints are converted somewhere else too, but maps, pids and other types are inserted in the list as they are. Then the list gets sent into a socket and it crashes in the `erlang:iolist_size` call.

So is there a reason why some types were omitted from the conversion into string? I can make a pull request, but maybe i'm missing something important about it.

**Comment by kaos** [comment]
No, it feels like an oversight. It would be better to make sure that its a valid iolist that is the result of a render (as far as feasible).
